### PR TITLE
Add 61 rotate linked list

### PR DESCRIPTION
**Pull request by oleggromov**
https://leetcode.com/problems/rotate-list

## My approach
We need to rotate a linked list `k` steps to the right. 

When I looked closer at "rotation", it merely meant to:
- connect the last node to the first one
- remove the connection between some in the middle and return the new head

So, in order to do that:
- I calculate the length of the list with aux method (`O(n)`). It can be done within the same loop but then it becomes messier. 
- See if k is larger than the length and use `k = k % length` in such case so we don't do unnecessary iterations
- Return right away if `k == 0` or the list is empth

Then,
- I proceed through array till then end (we need the last element)
- As the index becomes equal to the element I look for, I save it
- Break out of the loop when it's over

In the end I just have to change a few pointers and return the `new_head`.

WDYT about this?

### CODE REVIEW

File: 141-linked-list-cycle.py [hunk 1]
```diff
@@ -6,7 +6,27 @@
...
+    if slow == fast:
+      return True
+
+    slow = slow.next
+    if fast.next and fast.next.next:
+      fast = fast.next.next
+    else:
+      fast = None
+
+  return False
```
Comment (oleggromov) [inline 141-linked-list-cycle.py]: Disregard this. I messed up git's history a bit and don't really want to unravel it :)

File: 61-rotate-linked-list.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# class ListNode:
+#     def __init__(self, x):
+#         self.val = x
+#         self.next = None
+
+class Solution:
+  def rotateRight(self, head: ListNode, k: int) -> ListNode:
+    '''
+    Time complexity: O(2n)
+    Space complexity: O(1) ??
```
Comment (vitkarpov) [inline 61-rotate-linked-list.py]: Time is just `n` (`n` shows that it's just "linear" even if we may check the list 10 times) and space is `1` indeed. I believe we use the same amount of memory no matter how long the list is, don't we?

File: 141-linked-list-cycle.py (hunk 1 above)
Comment (vitkarpov) [reply 141-linked-list-cycle.py]: Okay 😄

File: 61-rotate-linked-list.py (hunk 2 above)
Comment (oleggromov) [reply 61-rotate-linked-list.py]: Yes, both is true indeed. Updated.

**Comment by oleggromov** [comment]
Thanks, @vitkarpov!

**Pull request merged**